src/test/resources/for-execution/operators.golo

```diff
@@ -77,3 +77,49 @@ function null_guarded = {
   let map = map[]
   return map: get("bogus") orIfNull "n/a"
 }
+
+function polymorphic_number_comparison = {
```

Comment:
Perhaps the checks could be enforced with a `raise` rather than just returning `res`?